### PR TITLE
Rename OMR_SCAVENGER_CACHE_* to OMR_COPYSCAN_CACHE_*

### PR DESCRIPTION
**Pull request by dmitripivkine**
Rename OMR_SCAVENGER_CACHE_* constants to more neutral OMR_COPYSCAN_CACHE_*. Usage of Copy Scan Caches is not limited to Scavenger and can be used by other Garbage Collectors.

### CODE REVIEW

**Comment by dmitripivkine** [comment]
@amicic Would you please review?

**Comment by dmitripivkine** [comment]
@babsingh Would you please proceed?

**Comment by dmitripivkine** [comment]
> There is a typo in the commit message: `neitral`.

fixed

**Comment by babsingh** [comment]
jenkins build all

**Comment by babsingh** [comment]
> https://dev.azure.com/eclipse-omr/ea4519db-b27e-4d19-a971-f01491f803e3/_apis/build/builds/7222/logs/45

It seems unrelated to this PR. Can you confirm if it is unrelated?
```
2023-03-08T17:02:18.0296580Z 32: [----------] 38 tests from PortSysinfoTest
2023-03-08T17:02:18.0326490Z 32: /Users/runner/work/1/s/fvtest/porttest/si.cpp line  691: omrsysinfo_test_sysinfo_set_limit_ADDRESS_SPACE omrsysinfo_set_limit set ADDRESS_SPACE soft max FAILED
2023-03-08T17:02:18.0412080Z 32: 
2023-03-08T17:02:18.0512590Z 32: 		LastErrorNumber: -113
2023-03-08T17:02:18.0595740Z 32: 		LastErrorMessage: Invalid argument
2023-03-08T17:02:18.0606350Z 32: 
2023-03-08T17:02:18.0707560Z 32: /Users/runner/work/1/s/fvtest/porttest/testHelpers.cpp:109: Failure
2023-03-08T17:02:18.0797930Z 32: Value of: 0 == numberFailedTestsInComponent
2023-03-08T17:02:18.0806100Z 32:   Actual: false
2023-03-08T17:02:18.0902600Z 32: Expected: true
2023-03-08T17:02:18.0908110Z 32: Test failed!
2023-03-08T17:02:18.1005310Z 32: [  FAILED  ] PortSysinfoTest.sysinfo_test_sysinfo_set_limit_ADDRESS_SPACE (0 ms)
2023-03-08T17:02:18.1009480Z 32: originalSoftLimit=65536
2023-03-08T17:02:18.1108020Z 32: originalHardLimit=65536
2023-03-08T17:02:18.1209930Z 32: soft set to hard limit=65536
```

**Comment by dmitripivkine** [comment]
yes, looks like this error is unrelated to the change

File: gc/base/standard/CopyScanCacheStandard.hpp
Comment (keithc-ca) [inline gc/base/standard/CopyScanCacheStandard.hpp:74]: Please use `OMR_ARE_ALL_BITS_SET()`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: gc/base/CopyScanCache.hpp
Comment (dmitripivkine) [reply gc/base/CopyScanCache.hpp:44]: > This sentence should end with a period.

this line is going to be removed in next PR, is there reason to address such nit-picks?

File: gc/base/CopyScanCache.hpp
Comment (dmitripivkine) [reply gc/base/CopyScanCache.hpp:104]: > This should use `OMR_ARE_ALL_BITS_SET()`.

Functions like this are not used in GC code, this is VM specific review requirement. I am not sure we should start to do it in this change

File: gc/base/standard/CopyScanCacheList.cpp
Comment (dmitripivkine) [reply gc/base/standard/CopyScanCacheList.cpp:137]: > Please use `OMR_ARE_ANY_BITS_SET()`, here and on line 168.

Functions like this are not used in GC code, this is VM specific review requirement. I am not sure we should start to do it in this change

File: gc/base/standard/CopyScanCacheStandard.hpp
Comment (dmitripivkine) [reply gc/base/standard/CopyScanCacheStandard.hpp:74]: > Please use `OMR_ARE_ALL_BITS_SET()`.

Functions like this are not used in GC code, this is VM specific review requirement. I am not sure we should start to do it in this change

File: gc/base/standard/CopyScanCacheList.cpp
Comment (dmitripivkine) [reply gc/base/standard/CopyScanCacheList.cpp:137]: They are not required by coding standards, are they?

File: gc/base/CopyScanCache.hpp
Comment (keithc-ca) [reply gc/base/CopyScanCache.hpp]: Only a committer could insist, but why should comments that don't use proper grammar and punctuation be acceptable?

**Comment by babsingh** [comment]
The last pending change is `OMR_ARE_ALL_BITS_SET`. There is a disagreement on this change. A discussion can happen on the side to resolve the disagreement. As part of the rename effort, this change is not necessary so we should not block this PR and prevent further development. My preference will be to merge the PR and address the disagreement on the side.

jenkins build all

**Pull request merged**
